javascript/transform.js

```diff
@@ -58,36 +62,24 @@ function gcj2wgs(gcjLat, gcjLng) {
 	if (outOfChina(gcjLat, gcjLng)) {
 		return {"lat": gcjLat, "lng": gcjLng};
 	}
-	var d = delta(gcjLat, gcjLng);
+	// since gcj \approx wgs, let's just use this one..
+	var d = gcjDelta(gcjLat, gcjLng);
 	return {"lat": gcjLat - d.lat, "lng": gcjLng - d.lng};
 }
 exports.gcj2wgs = gcj2wgs;
 
 function gcj2wgs_exact(gcjLat, gcjLng) {
-	var initDelta = 0.01;
-	var threshold = 0.000001;
-	var dLat = initDelta, dLng = initDelta;
-	var mLat = gcjLat-dLat, mLng = gcjLng-dLng;
-	var pLat = gcjLat+dLat, pLng = gcjLng+dLng;
-	var wgsLat, wgsLng;
+	var wgsLat = gcjLat, wgsLng = gcjLng;
+	var threshold = 0.000001; // ~0.55 m equator & latitude
+	
 	for (var i = 0; i < 30; i++) {
-		wgsLat = (mLat+pLat)/2;
-		wgsLng = (mLng+pLng)/2;
-		var tmp = wgs2gcj(wgsLat, wgsLng)
-		dLat = tmp.lat-gcjLat;
-		dLng = tmp.lng-gcjLng;
-		if ((Math.abs(dLat) < threshold) && (Math.abs(dLng) < threshold)) {
-			return {"lat": wgsLat, "lng": wgsLng};
-		}
-		if (dLat > 0) {
-			pLat = wgsLat;
-		} else {
-			mLat = wgsLat;
-		}
-		if (dLng > 0) {
-			pLng = wgsLng;
-		} else {
-			mLng = wgsLng;
+		var gcjDiff = gcjDelta(wgsLat, wgsLng);
```

Comment:
I will try to understand the R description during sprint festival. Thanks. If it's OK, I will try to merge it to other language.

Woof.